pkg/operator/oauthapiencryption/oauthapiencryptioncontroller.go

```diff
@@ -29,102 +27,98 @@ This annotation indicates that OAS-O manages this secret.`
+		// case 2: the encryptionCfg exists and it is annotated in the global namespace, we need to remove the annotation
+		if err := c.removeAnnotationAndRecordEvent(ctx, operatorclient.GlobalMachineSpecifiedConfigNamespace, oauthAPIServerEncryptionCfgInManagedNamespace.DeepCopy(), controllerContext.Recorder()); err != nil {
 			return err
 		}
-		controllerContext.Recorder().Eventf("SecretCreated", "Created %s in %s namespace because it was missing", oauthAPIServerEncryptionCfgName, operatorclient.GlobalMachineSpecifiedConfigNamespace)
-		return nil
 	}
-	if _, exist := oauthAPIServerEncryptionCfg.Annotations[EncryptionConfigManagedBy]; exist {
-		// case 2: exists and it is annotated
-		oauthEncryptionCfgData := oauthAPIServerEncryptionCfg.Data[encryptionconfig.EncryptionConfSecretKey]
-		oasEncryptionCfgData := openshiftAPIServerEncryptionCfg.Data[encryptionconfig.EncryptionConfSecretKey]
-		if !equality.Semantic.DeepEqual(oauthEncryptionCfgData, oasEncryptionCfgData) {
-			encryptionCfg := oauthAPIServerEncryptionCfg.DeepCopy()
-			encryptionCfg.Data[encryptionconfig.EncryptionConfSecretName] = oasEncryptionCfgData
-			_, err := c.secretClient.Update(ctx, encryptionCfg, metav1.UpdateOptions{})
-			if err != nil {
-				return err
-			}
-			controllerContext.Recorder().Eventf("SecretUpdated", "Updates %s in %s namespace because it was out of date with %s ", oauthAPIServerEncryptionCfgName, operatorclient.GlobalMachineSpecifiedConfigNamespace, openshiftAPIServerEncryptionCfg.Name)
-			return nil
-		}
-		return nil
+
+	oauthAPIServerEncryptionCfgInTargetNamespace, err := c.secretListerForOAuthTargetNamespace.Get(encryptionconfig.EncryptionConfSecretName)
+	if err != nil {
+		// this probably means that the encryption cfg hasn't been synced to the target namespace
+		return err
+	}
+	if _, exist := oauthAPIServerEncryptionCfgInTargetNamespace.Annotations[EncryptionConfigManagedBy]; exist {
+		// case 2.1: the encryptionCfg exists and it is annotated in the target (oauthAPIServerTargetNamespace) namespace, we need to remove the annotation
+		//
+		// note: this is nice to have since both operators watch the encryption config in the global namespace
+		return c.removeAnnotationAndRecordEvent(ctx, c.oauthAPIServerTargetNamespace, oauthAPIServerEncryptionCfgInTargetNamespace.DeepCopy(), controllerContext.Recorder())
 	}
 
-	// case 3: no-op the secret is managed by CAO
+	// case 3: no-op the encryption config is already managed by CAO
+	return nil
+}
+
+func (c *oauthEncryptionConfigSyncController) removeAnnotationAndRecordEvent(ctx context.Context, ns string, encryptionCfg *corev1.Secret, recorder events.Recorder) error {
+	delete(encryptionCfg.Annotations, EncryptionConfigManagedBy)
+	_, err := c.secretClient.Secrets(ns).Update(ctx, encryptionCfg, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+	recorder.Eventf("EncryptionConfigUpdated", "Removed the EncryptionConfigManagedBy annotation from %s in %s namespace", encryptionCfg.Name, ns)
```

Comment:
value of EncryptionConfigManagedBy is "EncryptionConfigManagedBy"?

Reply:
`EncryptionConfigManagedBy` = "encryption.apiserver.operator.openshift.io/managed-by"

Reply:
replace the string

Reply:
done.